Website Header + Navigation

CraaazySteeeve opened the pull request:
This PR includes:
A Header with:
- ATX Logo.
- Navigation options (only on Desktop, on mobile they are moved to the bottom of the screen)
- If the User isn't logged in:
- - A Twitch sign in button (Which isn't functional yet, as the login implementation will be done seperately)
- If the User IS logged in:
- - User currency/networth.
- - Username, and their PFP.
- - These user details are pull from a svelte store which could be updated with the details from the Twitch Login.

It DOES NOT include:
- The actual Twitch Login functionality.
- A daily login bonus button/functionality. (If it's decided this will be in the header, it can be added in a seperate ticket
- Signing out functionality. This will be depend on the implementation of the sign-in logic.

RubenOlano commented on frontend/src/lib/components/HeaderNavigationButton.svelte:
```diff
@@ -0,0 +1,14 @@
+<a href={link}>
+    <div class="flex flex-col items-center text-zinc-300 hover:text-white p-1 pt-2 text-xl">
+        <Fa icon={faIcon} />
+        <span class="text-sm">{label}</span>
```
I'd recommend hiding text whenever in the mobile view since we're already using icons. Something like `class="hidden md:block text-sm"` in order to hide on smaller view

RubenOlano commented on frontend/src/lib/components/HeaderUserView.svelte:
```diff
@@ -0,0 +1,58 @@
+        </div>
+    </button>
+    <button on:click={toggleDropdown} type="button" class="sm:hidden text-gray-500 hover:text-white focus:text-white focus:outline-none">
+        <svg class="h-8 w-8 fill-current" viewBox="0 0 24 24">
+            <path class="{isDropdownVisible == true ? "block" : "hidden"}" fill-rule="evenodd" d="M18.278 16.864a1 1 0 0 1-1.414 1.414l-4.829-4.828-4.828 4.828a1 1 0 0 1-1.414-1.414l4.828-4.829-4.828-4.828a1 1 0 0 1 1.414-1.414l4.829 4.828 4.828-4.828a1 1 0 1 1 1.414 1.414l-4.828 4.829 4.828 4.828z"/>
+            <path class="{isDropdownVisible == true ? "hidden" : "block"}" fill-rule="evenodd" d="M4 5h16a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2z"/>
+        </svg>
```
Extract SVG into a searate component so its easier to tell what is going on

sakulstar replied:
I don't think splitting them into separate SVG would simplify it, though I have noticed that changing the second value from true to false makes it less confusing, so I have committed those changes.

RubenOlano commented on frontend/src/lib/components/MobileNavigationButton.svelte:
```diff
@@ -0,0 +1,14 @@
+<a href={link}>
+    <div class="flex flex-col items-center text-zinc-300 bg-blue-900 hover:bg-blue-600 rounded p-1 pt-2 text-xl w-20">
+        <Fa icon={faIcon} />
+        <span class="text-xs">{label}</span>
```
Something like class="hidden md:block text-sm" in order to hide on smaller view

sakulstar replied:
[Explained it above](https://github.com/AtriEx/ATX/pull/9#discussion_r1480040913)

RubenOlano commented on frontend/src/lib/components/CurrencyView.svelte:
```diff
@@ -0,0 +1,10 @@
+<script lang="ts">
+	import Fa from "svelte-fa";
+    import {faCoins} from '@fortawesome/free-solid-svg-icons'
+    import {currency} from '$lib/stores/userData.js';
```
Minor nitpick but idk why this is importing as a `js` file rather than `ts`

CraaazySteeeve replied:
Resolved, removed 'js' extension.

RubenOlano commented on frontend/src/lib/components/HeaderUserView.svelte:
```diff
@@ -0,0 +1,58 @@
+    import CurrencyView from "./CurrencyView.svelte";
+	import NetworthView from "./NetworthView.svelte";
+
+    import { isLoggedIn, username, userPfp} from '$lib/stores/userData.js';
```
Some here about `js` and `ts`

CraaazySteeeve replied:
Resolved, removed 'js' extension.

RubenOlano reviewed (changes requested):
I'd personally recommend removing the text from smaller screen buttons for navigation since they are self-explanatory. However you are free to overrule me on that call

CraaazySteeeve commented:
> I'd personally recommend removing the text from smaller screen buttons for navigation since they are self-explanatory. However you are free to overrule me on that call

I agree. I have modified the mobile buttons to be bigger, and removed the text.
